### PR TITLE
Implementing multiversion docs

### LINKED ISSUE #276: Multiversion docs are missing
The legacy reader was never adapted. We should do this.

Pinging @ansys/pyansys-core - if anybody can take the lead here, great. Otherwise I'll try to find some time (down the road)

### PR DESCRIPTION
**Pull request by clatapie**
Closes #276.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,55 +1,42 @@
...
-
-- repo: https://github.com/pre-commit/mirrors-clang-format
-  rev: v16.0.6
-  hooks:
-  - id: clang-format
-    files: |
-      (?x)^(
-        ansys/mapdl/reader/cython/[^_].*\.c.*
-      )$
-
```
Comment (RobPasMue) [inline .pre-commit-config.yaml:43]: Keep this hook @clatapie

File: doc/source/conf.py [hunk 2]
```diff
@@ -151,6 +151,10 @@
     "additional_breadcrumbs": [
         ("PyAnsys", "https://docs.pyansys.com/"),
     ],
+    "switcher": {
+        "json_url": f"https://{cname}/versions.json",
+        "version_match": switcher_version,
+    },
 }
```
[the remark this answers is not in the document]
Comment (RobPasMue) [reply doc/source/conf.py:159]: As well as edit page buttons. Please add it

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: ansys/mapdl/reader/__init__.py [hunk 3]
```diff
@@ -3,21 +3,12 @@
...
-    Archive,
-    save_as_archive,
-    write_cmblock,
-    write_nblock,
-)
-from ansys.mapdl.reader.cell_quality import quality
-from ansys.mapdl.reader.common import read_binary
-from ansys.mapdl.reader.misc import Report, _configure_pyvista
-
-from . import _archive
```
[the remark this answers is not in the document]
Comment (clatapie) [reply ansys/mapdl/reader/__init__.py:20]: It was not used by the file

**Comment by RobPasMue** [comment]
Also, @clatapie - implementing Vale might be a burden really... let's just remove it from the workflow

**Comment by RobPasMue** [comment]
> Also, @clatapie - implementing Vale might be a burden really... let's just remove it from the workflow

It wasn't present before either

**Comment by RobPasMue** [comment]
Also... you removed the ``code-style``. That one should still be run. The one that you should remove is the ``doc-style`` stage.

**Comment by RobPasMue** [comment]
I'm not sure it's a good idea removing the current ``setup.py``: it was doing a lot of different things for compilation of CPython modules + it's not really needed to switch to multiversion docs. Let's try to stay within the scope of the PR. Our GitHub actions work as well with a requirements folder containing requirements_*.txt files.

**Comment by RobPasMue** [comment]
Closing in favor of #278

**Pull request closed without merging**